### ISSUE TITLE
Restarting Redux environment without triggering preventDoubleInitialization?

### ISSUE THREAD

**Issue by Slapbox** [klarna/electron-redux#293]
We just upgraded to the `2.0.0-alpha.9` and the new setup looks amazingly simple!

I've not gotten things working quite yet, and maybe won't be able to quite yet, but one thing I noticed is `preventDoubleInitialization`

https://github.com/klarna/electron-redux/blob/alpha/src/utils/misc.ts#L5

With electron-redux@1.x we had built in a custom functionality to restart the redux environment to avoid having to restart our entire project.

The core of our solution was just to use something like below to unregister all channels before we re-create our app's windows from scratch:

```js
    channels.forEach(channel => {
      ipcMain.removeAllListeners(channel);
    });
 ```
 
 Is there a way to achieve this with the new architecture? If not, might I suggest that `previouslyInitialized` (https://github.com/klarna/electron-redux/blob/alpha/src/utils/misc.ts#L5) be made into a global instead so users can manually reset it if needed?

**Comment by Slapbox** [comment]
Turns out this is still very manageable although the solution forces us into using inline `require`s rather than `import` at the top of our file. It's a minor gripe, but it would be great to have this use case built into the architecture.

**configurestore.js**
Use inline requires to work around the `previouslyInitialized` variable.
```js
  if (scope === 'renderer') {
    enhancers.push(require('electron-redux').rendererStateSyncEnhancer());
  } else if (scope === 'main') {
    enhancers.push(require('electron-redux').mainStateSyncEnhancer()); 
  }
  ```
  
  **main.js**
Remove the IPC handler to prevent an Electron error about double registered handlers
  ```js
  
  ipcMain.removeHandler('electron-redux.INIT_STATE_ASYNC');
  
  ```

**Comment by Slapbox** [comment]
@matmalkowski any idea how we might make this work better? It was working for some time when we did the above in our `main` process, but increasingly often we're getting:

`Error: electron-redux has already been attached to a store`

Getting this working with `v1` was trivial and it would be awesome if it could be made that simple in `v2`. Project restarts are really slowing us down now, since every refresh is turning into an error requiring a full restart which takes about two minutes.

**Comment by matmalkowski** [comment]
@Slapbox I think I saw a notification yesterday at night about a comment, but I don't see that comment here anymore 🤔 

In terms of supporting this - I don't see any reason why we could not extend API to support this in the future - but we would need to understand how we plan to solve that and so on. If you are interested and have some ideas in mind, let me know and we could brainstorm it together

**Comment by Slapbox** [comment]
Indeed, I commented but I found the issue. But I'm not really sure why removing some code I'd committed fixed it.

Would there be any downside to making `previouslyInitialized` a global so that users could manually reset it? Or otherwise what would you think of a function like, `resetPreviouslyInitialized()` to set it back to false?

The solution we have right now involves using inline requires which are a bit more difficult to reason about I feel.
